docs/book/beyond_basics/dependencies.md

```diff
@@ -0,0 +1,178 @@
+
+Before version 1.0.2, Kubebuilder shipped a pre-baked manifest of the
+correct dependencies.  When scaffolding out at new project using
+`kubebuilder init` (a **kb-project**), it would copy over a `Gopkg.toml`
+file containing the exact dependency versions required for the project
+(which could then be used by `dep` dependency management tool to actually
+fetch the dependencies).
+
+In addition to the Kubernetes dependencies required, this also specified
+that all kb-projects used the master branch of the **controller-runtime**
+library, which provides the abstractions that Kubebuilder is built upon.
+Because controller-runtime wraps and consumes Kubernetes, it *also* needs
+specific versions of the k8s-deps, and those version *must* match the ones
+listed in the kb-project's Gopkg.toml, otherwise we'd have conflicting
+dependencies.
+
+#### The Problem with Prebaked Manifests
+
+Using the master branch as the target version of controller-runtime made
+it impossible to make breaking changes to controller-runtime.  However,
+even when using a specific version of controller-runtime, it's still
+difficult to make changes.
+
+Since kb-projects must use an identical set of dependencies to
+controller-runtime, any update to the controller-runtime dependencies
+(say, to pull in a new feature) would have caused immediate dependency
+version conflicts.  Effectively, any update to the dependencies had to be
+treated as a major version revision, and there would have been no way to
+tell the difference between "this release includes breaking API changes"
+and "this release simply switches to a newer version of the k8s-deps".
+
+### Transitive Dependencies (Kubebuilder 1.0.2+)
+
+As noted above, any dependency version in kb-projects must match
+dependency versions listed in controller-runtime, exactly. Furthermore, it
+turns out, by design, the set of k8s-deps used in controller-runtime is
+a superset of the set of dependencies actually imported by kb-projects.
+
+Therefore, in kb-projects generated with Kubebuilder 1.0.2+, no
+dependencies are listed besides controller-runtime (and controller-tools).
+All of the k8s-deps become transitive dependencies, whose versions are
+determined when `dep` (the dependency management tool) looks at the
+versions required by controller-runtime.
+
+controller-runtime is semantically versioned, so any changes to either the
+interfaces in controller-runtime, or the pieces of the k8s-deps that are
+exposed as part of those interfaces, means a new major version of
+controller-runtime will be released.  Any other changes (new features, bug
+fixes, updates to k8s-deps which don't break interfaces) yield minor or
+patch versions (as per [semver](https://semver.org)), which can easily and
+safely be updated to by kb-projects.
+
+#### controller-tools Dependencies
+
+controller-tools, the library used to generate CRD and RBAC manifests for
+kb-projects, does not directly depend on controller-runtime, but shares
```

Comment:
currently controller-tools *does* depend on controller-runtime because it has `test` projects which uses controller-runtime. Once we remove those deps, it should be safe to say so. Infact, while doing this exercise, I realized, whole of scaffold pkgs should be moved from the controller-tools to Kubebuilder. We should just keep controller-gen tools in controller-tools.